feature/onerepmax/src/main/kotlin/com/patrykandpatryk/liftapp/feature/onerepmax/viewmodel/OneRepMaxViewModel.kt

```diff
@@ -0,0 +1,74 @@
+package com.patrykandpatryk.liftapp.feature.onerepmax.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.patrykandpatryk.liftapp.core.extension.smartToFloatOrNull
+import com.patrykandpatryk.liftapp.core.extension.smartToIntOrNull
+import com.patrykandpatryk.liftapp.domain.repository.PreferenceRepository
+import com.patrykandpatryk.liftapp.feature.onerepmax.model.OneRepMaxUiState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class OneRepMaxViewModel @Inject constructor(
+    preferenceRepository: PreferenceRepository,
+) : ViewModel() {
+
+    val massUnit = preferenceRepository.massUnit.get()
+
+    var oneRepMaxUiState by mutableStateOf(
```

Comment:
It would be great to save the state in `SavedStateHandle`, but this improvement can be added later.

Reply:
Please see b724bfd623626eb2176b78363b3c174ba6f7142b.